org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/breakpoints/LambdaLocationLocator.java

```diff
@@ -67,10 +73,20 @@ public boolean visit(LambdaExpression node) {
 		fNodeOffset = node.getStartPosition();
 		IMethodBinding methodBinding = node.resolveMethodBinding();
 		if (methodBinding != null) {
-			fLambdaMethodName = LambdaLocationLocatorHelper.toMethodName(methodBinding);
-			fLambdaMethodSignature = LambdaLocationLocatorHelper.toMethodSignature(methodBinding);
-			fLocationFound = true;
+			fCount++;
+			if (fCount == fLambdaPosition || fLambdaPosition == -10) {
```

Comment:
`-10` is a no go. This seem to be a "special" value - so please add a constant with good name what kind of "special" value it is.